ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,141 @@ def test_trustdomain_disable_disables_subdomain(self):
+        so that parent child relationship is as follows:
+        "a"->"b"->"c"
+
+        user1 is direct member of "a" and as a result
+        member of "b" and "c"".
+        Now if one adds a direct membership to "b"
+        nothing will change.
+
+        Now if one removes the direct membership to "b"
+        nothing should change, the memberships should be honored
+        Linked Issue: https://pagure.io/SSSD/sssd/issue/3636
+        """
+        master = self.master
+        client = self.clients[0]
+        tasks.kinit_admin(master)
+
+        # add hbacrule
+        master.run_command(['ipa', 'hbacrule-add', '--usercat', 'all',
+                            '--hostcat', 'all', '--servicecat', 'all',
+                            'any_to_any'])
+        # add a user
+        username = "testuser" + str(random.randint(200000, 9999999))
+        userpasswd = 'Secret123'
+
+        input_passwd = '{upasswd}\n{upasswd}\n'.format(upasswd=userpasswd)
+        tasks.user_add(master, username, password=userpasswd)
+
+        cmd_output = master.run_command(['ipa', 'passwd', username],
+                                        stdin_text=input_passwd)
+        assert 'Changed password' in cmd_output.stdout_text
+
+        # add group a
+        cmd_output = master.run_command(['ipa', 'group-add', 'a'])
+        assert 'Added group "a"' in cmd_output.stdout_text
+
+        # add group b
+        cmd_output = master.run_command(['ipa', 'group-add', 'b'])
+        assert 'Added group "b"' in cmd_output.stdout_text
+
+        # add group c
+        cmd_output = master.run_command(['ipa', 'group-add', 'c'])
+        assert 'Added group "c"' in cmd_output.stdout_text
+
+        # add group members
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'b', '--groups', 'a'])
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'c', '--groups', 'b'])
+        assert 'Group name: c' in cmd_output.stdout_text
+        assert 'Member groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member groups: a' in cmd_output.stdout_text
+
+        # add user to group 'a'
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'a', '--users', username])
+        assert 'Group name: a' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member of groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member of group: c' in cmd_output.stdout_text
+
+        # clear sssd_cache
+        clear_sssd_cache(master)
+
+        # user lookup
+        cmd = 'ipa user-show {} | grep group'.format(username)
```

Comment:
We don't really need `|` here. We are printing only 1 user details and that would be much lines. Please remove it.

Reply:
I only want to view and then assert on the  group info, rest of the info is not of use here.

Reply:
fixed it